routing/payment_lifecycle.go

```diff
@@ -247,7 +247,19 @@ func (p *paymentLifecycle) resumePayment() ([32]byte, *route.Route, error) {
 
 		// We found a route to try, launch a new shard.
 		attempt, outcome, err := shardHandler.launchShard(rt)
-		if err != nil {
+		switch err {
+		// We can get an already failed error in the case where we
```

Comment:
So now we'll go back to the top, and exit as we now have a failure (`state.terminate==True`) and there're no more active shards?

Reply:
Yep exactly, we're def in a terminate state so now we just need to wait for our shards to come back (if any). 

Wondering whether we should `waitForShard` for each of our `numActiveShards` here? Otherwise we'll loop through checkshards/ (pointless) pathfinding / err already failed while we wait for remaining shards to come back and we hit `state.terminate && state.numShardsInFlight == 0`.

Reply:
won't we go into the second case `state.terminate || state.remainingAmt == 0` which will cal `waitForShards`?